src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy

```diff
@@ -433,15 +588,26 @@ public enum MysteriousTreasures implements LogicCardInfo {
             energyCost R, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
+              sandAttack(thisMove)
             }
           }
           move "Flame Drum", {
             text "80 damage. If Magby isn’t anywhere under Magmortar, discard 2 Energy cards from your hand. (If you can’t discard 2 Energy cards from your hand, this attack does nothing.)"
             energyCost R, R, C
-            attackRequirement {}
+            attackRequirement {
+              def condition1 = (self.getPokemonCards().any{it.name.contains("Magby")})
+              def condition2 = (my.hand.filterByType(ENERGY).size() >= 2)
+              assert (condition1 || condition2) : "Magby isn't anywhere under Magmortar, and you can't discard 2 Energy cards from your hand"
```

Comment:
The text doesn't quite match the check which may cause confusion.

Reply:
It does? Seems good to me 🤔 
* Card Text: _"**If Magby isn’t anywhere under Magmortar**, discard 2 Energy cards from your hand. (**If you can’t discard 2 Energy cards from your hand, this attack does nothing.**)"_

As long as any of those two conditions are met, it should be able to attack. I did alter the attack though, since it wasn't doing damage when Magby was under Magmortar. But don't think that's what you meant, could you let me know where in the assert do you think it's not quite matching?

Reply:
I mean the assertion text. Code uses an OR, text is an AND. Along with that "Magby isn't anywhere under Magmortar" is the opposite of what condition 1 is actually checking. If somebody evolved from Magby and tried using it, the text would tell them "Magby isn't anywhere under Magmortar" even though it is. It would instead be blaming them having not enough energy to discard even though they most likely do.

Reply:
I don't see the code doing that. The two conditions are for different scenarios, and second doesn't need to be checked at all if Magmortar has a Magby below (which is why I used an OR for the assert):
* Magby is below Magmortar, assert is OK.
* Magby is not below Magmortar, but there are energies, assert is OK
* Both things don't happen, throw the assert fail.

Reply:
I'll just blame it being 3AM at the time, let's be honest. I see what's going on now.